pkg/vm/engine/tae/containers/pool.go

```diff
@@ -55,18 +58,51 @@ func WithFixedSizeRatio(ratio float64) VectorPoolOption {
 	}
 }
 
+func WithMPool(mp *mpool.MPool) VectorPoolOption {
```

Comment:
Do we really need this another level of indirection?

Reply:
Maybe it can be a required parameter for NewVectorPool.